app/contracts/work_packages/base_contract.rb

```diff
@@ -335,28 +347,111 @@ def validate_remaining_work_is_lower_than_work
       end
     end
 
+    # to be removed in 15.0 with :percent_complete_edition feature flag removal
     def validate_remaining_work_is_set_when_work_is_set
       if work_set? && !remaining_work_set?
         errors.add(:remaining_hours, :must_be_set_when_work_is_set)
       end
     end
 
+    # to be removed in 15.0 with :percent_complete_edition feature flag removal
     def validate_work_is_set_when_remaining_work_is_set
       if remaining_work_set? && !work_set?
         errors.add(:estimated_hours, :must_be_set_when_remaining_work_is_set)
       end
     end
 
+    def validate_work_is_set_when_remaining_work_and_percent_complete_are_set
+      if remaining_work_set_and_valid? && percent_complete_set_and_valid? && work_unset?
+        errors.add(:estimated_hours, :must_be_set_when_remaining_work_and_percent_complete_are_set)
+      end
+    end
+
+    def validate_remaining_work_is_set_when_work_and_percent_complete_are_set
+      if work_set_and_valid? && percent_complete_set_and_valid? && remaining_work_unset?
+        errors.add(:remaining_hours, :must_be_set_when_work_and_percent_complete_are_set)
+      end
+    end
+
+    def validate_percent_complete_is_set_when_work_and_remaining_work_are_not_set
+      if work_set_and_valid? && remaining_work_set_and_valid? && work != 0 && percent_complete_unset?
+        errors.add(:done_ratio, :must_be_set_when_work_and_remaining_work_are_set)
+      end
+    end
```

Comment:
Is this method name accurate to its contents?

"Percent complete is set when work and remaining work aren't"

but the method's checks indicate it's checking for work and remaining work both being set

Reply:
You're totally right. Good catch 👍